x/twap/logic_test.go

```diff
@@ -539,316 +537,6 @@ func (s *TestSuite) TestGetInterpolatedRecord_ThreeAsset() {
-		},
-		// TODO: overflow tests
-		// - max spot price
-		// - large time delta
-		// - both
-
-		// TODO: hand calculated tests
-	}
-
-	for name, tc := range tests {
-		t.Run(name, func(t *testing.T) {
-			osmoassert.ConditionalPanic(t, tc.expPanic, func() {
-				actualTwap := twap.ComputeGeometricTwap(tc.startRecord, tc.endRecord, tc.quoteAsset)
-				osmoassert.DecApproxEq(t, tc.expTwap, actualTwap, osmomath.GetPowPrecision())
-			})
-		})
-	}
-}
-
-// TODO: split up this test case to cover both arithmetic and geometric twap
-func TestComputeArithmeticTwap_ThreeAsset(t *testing.T) {
-	testThreeAssetCaseFromDeltas := func(startAccum, accumDiff sdk.Dec, timeDelta time.Duration, expectedTwap sdk.Dec) computeThreeAssetArithmeticTwapTestCase {
-		return computeThreeAssetArithmeticTwapTestCase{
-			newThreeAssetOneSidedRecord(baseTime, startAccum, true),
-			newThreeAssetOneSidedRecord(baseTime.Add(timeDelta), startAccum.Add(accumDiff), true),
-			[]string{denom0, denom0, denom1},
-			[]sdk.Dec{expectedTwap, expectedTwap, expectedTwap},
-			false,
-		}
-	}
-
-	tenSecAccum := OneSec.MulInt64(10)
-	pointOneAccum := OneSec.QuoInt64(10)
-	tests := map[string]computeThreeAssetArithmeticTwapTestCase{
-		"three asset basic: spot price = 1 for one second, 0 init accumulator": {
-			startRecord: newThreeAssetOneSidedRecord(baseTime, sdk.ZeroDec(), true),
-			endRecord:   newThreeAssetOneSidedRecord(tPlusOne, OneSec, true),
-			quoteAsset:  []string{denom0, denom0, denom1},
-			expTwap:     []sdk.Dec{sdk.OneDec(), sdk.OneDec(), sdk.OneDec()},
-		},
-		"three asset same record: asset1, end spot price = 1": {
-			startRecord: newThreeAssetOneSidedRecord(baseTime, sdk.ZeroDec(), true),
-			endRecord:   newThreeAssetOneSidedRecord(baseTime, sdk.ZeroDec(), true),
-			quoteAsset:  []string{denom1, denom2, denom2},
-			expTwap:     []sdk.Dec{sdk.OneDec(), sdk.OneDec(), sdk.OneDec()},
-		},
-		"three asset. accumulator = 10*OneSec, t=5s. 0 base accum": testThreeAssetCaseFromDeltas(
-			sdk.ZeroDec(), tenSecAccum, 5*time.Second, sdk.NewDec(2)),
-
-		// test that base accum has no impact
-		"three asset. accumulator = 10*OneSec, t=5s. 10 base accum": testThreeAssetCaseFromDeltas(
-			sdk.NewDec(10), tenSecAccum, 5*time.Second, sdk.NewDec(2)),
-		"three asset. accumulator = 10*OneSec, t=100s. .1*second base accum": testThreeAssetCaseFromDeltas(
-			pointOneAccum, tenSecAccum, 100*time.Second, sdk.NewDecWithPrec(1, 1)),
-	}
-	for name, test := range tests {
-		t.Run(name, func(t *testing.T) {
-			for i, startRec := range test.startRecord {
-				actualTwap, err := twap.ComputeTwap(startRec, test.endRecord[i], test.quoteAsset[i], twap.ArithmeticTwapType)
-				require.Equal(t, test.expTwap[i], actualTwap)
-				require.NoError(t, err)
-			}
-		})
-	}
-}
-
-// This tests the behavior of computeArithmeticTwap, around error returning
-// when there has been an intermediate spot price error.
-func TestComputeArithmeticTwapWithSpotPriceError(t *testing.T) {
```

Comment:
We should keep this test case here and refactor it to take in strategy as an arg.  Spot price errors may occur for both geometric and arithmetic twaps